services/121-service/src/notifications/services/message.service.ts

```diff
@@ -167,7 +167,14 @@ export class MessageService {
     if (!existingPendingMessage) {
       return;
     }
-    await this.whatsappPendingMessageRepo.delete(pendingMessageId);
+
+    // Try to delete the pending message, if it was already deleted by another worker we do not need to process it
+    // This should solve any possible as record in a db cannot be delete twice
+    const deleteResult =
+      await this.whatsappPendingMessageRepo.delete(pendingMessageId);
+    if (deleteResult?.affected !== 1) {
```

Comment:
```suggestion
    if (deleteResult?.affected === 0) {
```

Feels safer? I wonder if `affected` could be > 1. But maybe I'm wrong.

Reply:
Ids are unique in a table due to a database constaint. I think I tried your way but I think I got a type-error as `affected` is potentially undefined (if you would use typeorm with a different underlying db than postgres)